ttnn/cpp/ttnn/operations/uniform/device/uniform_device_operation.cpp

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "uniform_device_operation.hpp"
+
+namespace ttnn::operations::uniform {
+
+UniformDeviceOperation::program_factory_t UniformDeviceOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    return Factory{};
+}
+
+void UniformDeviceOperation::validate_inputs(
+    const operation_attributes_t& attributes, const tensor_args_t& tensor_args) {
+    TT_FATAL(tensor_args.input_tensor.storage_type() == StorageType::DEVICE, "Input tensor need to be on device");
+    TT_FATAL(tensor_args.input_tensor.buffer() != nullptr, "Input tensor need to be allocated in buffers on device");
+    TT_FATAL((tensor_args.input_tensor.get_layout() == Layout::TILE), "Input tensor must be tilized");
+    TT_FATAL(
+        tensor_args.input_tensor.get_dtype() == DataType::BFLOAT16 ||
+            tensor_args.input_tensor.get_dtype() == DataType::FLOAT32,
+        "Input tensor must be Float32 or Bfloat16");
+    TT_FATAL(attributes.from <= attributes.to, "from param must be <= to");
+}
```

Comment:
Please add prefix for TT_FATAL messages.
Example: "Uniform: input tensor need to be on device".

Reply:
Updated